Automattic/WP-Job-Manager
# Deprecates Mixtape REST API implementation and enables core REST API

Fixes #1625

#### Changes proposed in this Pull Request:

* See issue for discussion.
* Enables `show_in_rest` for CPT and taxonomies.
* Throw deprecation notices (both in REST API header and with `trigger_error()`) for people with `WPJM_REST_API_ENABLED` constant set.

#### Testing instructions:
* Make sure Mixtape endpoints load properly and errors thrown with this set in `wp-config.php`:
`define( 'WPJM_REST_API_ENABLED', true );`
   * `GET http://wpjm.docker/wp-json/wpjm/v1/status`
   * `GET http://wpjm.docker/wp-json/wpjm/v1/settings`
* With that line removed or set to `false`, make sure:
  * Errors not shown or logged.
  * Endpoints from above return `404`.
  * Core endpoints work for `job_listings` and taxonomies.
     * `GET http://wpjm.docker/wp-json/wp/v2/job-listings`
     * `GET http://wpjm.docker/wp-json/wp/v2/job-types`
     * `GET http://wpjm.docker/wp-json/wp/v2/job-categories`

#### Proposed changelog entry for your changes:
* Deprecate custom REST API endpoints and move to a pure WP core REST API implementation. 

### Todo
